### PR TITLE
add metrics

### PR DESCRIPTION
**Pull request by Stargazer1998**
postitive / negative persistence

## Describe your changes
Positive and negative persistence added
## What is the related task number?
3.2
## Checklist before requesting a review
- [x] I have performed a self-review of my code
- [x] If it is a core feature, I have added thorough tests.

### CODE REVIEW

File: src/tum_dlr_automl_for_eo/utils/custom_metrics.py [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+
+    return pos_persistence
+
+
+def negative_persistence(acc_list):
+    neg_persistence = 1
+    number_of_models, number_of_epochs = np.shape(acc_list)
+    top25_len = number_of_models / 4
+    top25_len = int(np.ceil(top25_len))
+    neg_previous_top25 = np.argsort(acc_list[:, 0])[:top25_len]
```
Comment (emreds) [inline src/tum_dlr_automl_for_eo/utils/custom_metrics.py]: Why we are having the 25 here?

File: tests/unit/custom_metric_test.py [hunk 2]
```diff
@@ -0,0 +1,208 @@
+import numpy as np
+
+from src.tum_dlr_automl_for_eo.utils import custom_metrics
+
+random_acc_list = np.array([
+    [17, 74, 77, 81, 61, 84, 10, 82, 56, 66, 44, 40, 72, 23, 6, 59,
+     98, 34, 26, 5],
+    [15, 67, 3, 47, 8, 82, 36, 81, 29, 37, 37, 95, 67, 6, 44, 4,
+     79, 27, 12, 60],
+    [30, 57, 47, 29, 23, 92, 93, 90, 55, 95, 59, 38, 1, 93, 55, 97,
```
Comment (emreds) [inline tests/unit/custom_metric_test.py:10]: Actually, test cases doesn't need to be so long. They are just for demonstration, is it related with positive and negative resistance?

File: src/tum_dlr_automl_for_eo/utils/custom_metrics.py (hunk 1 above)
Comment (Stargazer1998) [reply src/tum_dlr_automl_for_eo/utils/custom_metrics.py]: To indicate that we are looking for best(worst) 25%

File: tests/unit/custom_metric_test.py (hunk 2 above)
Comment (Stargazer1998) [reply tests/unit/custom_metric_test.py:10]: We need to track whole all models at each epoch to see best(worst) 25%. So, using less models would not make sense to test the functions.

**Review by emreds — APPROVED** [review]

**Pull request merged**